pkg/oauth/providers/github.go

```diff
@@ -31,18 +30,18 @@ func getUserFromToken(c *gin.Context, token *oauth2.Token) (*models.UserCredenti
 
 	currTime := time.Now()
 	user := models.UserCredentials{
-		Name:         githubUser.Name,
+		Name:         githubUser.GetName(),
 		Kind:         models.GithubAuth,
 		Role:         models.RoleUser,
 		State:        models.StateActive,
 		SocialAuthID: githubUser.ID,
-		LoggedIn:     &types.TrueValue,
+		LoggedIn:     true,
 		CreatedAt:    &currTime,
 	}
 
 	for _, githubUserEmail := range githubUserEmails {
-		if *githubUserEmail.Primary {
-			user.Email = githubUserEmail.Email
+		if githubUserEmail.Primary != nil && githubUserEmail.Email != nil {
+			user.Email = githubUserEmail.GetEmail()
```

Comment:
Why this change?

Reply:
The package `go-github` that we are using has the `Email` field as `*string` and `user.Email` is converted to a `string`, so to get the value of email `go-github` provides this function

Reply:
Got it !!